mlir/lib/Dialect/Vector/Transforms/LowerVectorContract.cpp

```diff
@@ -458,49 +450,70 @@ struct UnrolledOuterProductGenerator
     return res;
   }
 
+  std::optional<int64_t> getReductionSize(VectorType vecType,
```

Comment:
Can you add a doc to the method? Otherwise, people need to look at implementation to understand when `std::nullopt` will be returned.